### ISSUE TITLE
Developer tools

### ISSUE THREAD

**Issue by olemartinorg** [Altinn/app-frontend-react#1021]
### Description

We're creating a platform used by developers to create apps. Let's give them tools.

I've had this marinating in the back of my head for a while, so when the opportunity arose to write a list of features I'd want on Slack, it's probably best to detail this in an issue as well.

### Some nice-to have features
- Visualizing the data model while filling out the form. There could be multiple ways to show the data model.
  - Resolved bindings for each component, aka `node.getFormData()` (although that does not currently work well with a `list` binding)
  - The entire data model in dot-notation (aka the redux state)
  - The entire data model as JSON (aka what we send off to save on the backend)
  - The entire data model as XML (we could probably ask the backend for the raw data)
  - See #136
- Functionality to inspect properties on all components
  - See expressions and what they resolve to, for each component in the form
  - Bonus points if we can visualize every evaluation step in an expression (in `["equals", ["component", "something"], "yes"]` we evaluate two discrete expression functions, first `["component", "something"]` and `["equals", "value", "yes"]`)
- Inspect the data model targets for every component binding
  - I've made a prototype for this, see #990
- Showing/hiding hidden components, and why they are hidden
- Showing/hiding hidden pages, and why they are hidden
- Showing errors/warnings (from expression evaluation and from validations as defined in #648)
- Navigating the current form pages without having to add a navigation bar while developing
- One button-click to preview PDF
- Show indicators to see which components trigger validations (and what will be validated when) in the GUI
- Adding functionality to load other versions of `app-frontend-react` (it should be possible to implement with some trickery). That means we could easily switch the version in tt02 and production as well.
- Keyboard shortcut to enable/display the developer tools, and alternatively another way to open them on mobile.
- Show the developer tools (or just a button for it) automatically when the domain is `local.altinn.studio`. That way we don't even have to advertise that we're shipping developer tools to app developers; they'll find out. Just make sure to inform them about the keyboard shortcut in production as well.

**Comment by bjosttveit** [comment]
Could it be an idea to add a "console"? The "real console" gets a bit cluttered with random warnings and such. For example, expressions log pretty errors to the console if they fail to resolve, maybe we could have a cleaner console in the developer tools with only intentional logs?

**Comment by bjosttveit** [comment]
Yeah, maybe just a log, we dont need to be able to send commands really. Maybe hard to make it display things correctly?
